src/transformers/models/trillsson_efficientnet/modeling_trillsson_efficientnet.py

```diff
@@ -0,0 +1,688 @@
+        return hidden_states
+
+
+class TrillssonInvertedResidual(nn.Module):
+    def __init__(
+        self,
+        config: TrillssonEfficientNetConfig,
+        in_channels: int,
+        out_channels: int,
+        stride: int,
+        expand_ratio: int,
+        use_se: int,
+        survival_probability: int,
+    ) -> None:
+        """TrillssonInvertedResidual block: Inverted Residual Bottleneck."""
+        super().__init__()
+        hidden_dim = round(in_channels * expand_ratio)
+        self.config = config
+        self.expand_ratio = expand_ratio
+        self.use_se = use_se
+        self.survival_probability = survival_probability
+        self.identity = stride == 1 and in_channels == out_channels
+        self.activation = ACT2FN[config.hidden_act]
+        self.dropout = nn.Dropout(self.survival_probability)
+        if self.expand_ratio != 1:
+            self.expand_conv = Conv2DSamePadding(
+                in_channels=in_channels, out_channels=hidden_dim, kernel_size=1, stride=1, padding=0, bias=False
+            )
+            self.expand_bn = nn.BatchNorm2d(
+                num_features=hidden_dim, eps=self.config.norm_eps, momentum=self.config.norm_momentum
+            )
+        self.dwconv2 = Conv2DSamePadding(
+            in_channels=hidden_dim,
+            out_channels=hidden_dim,
+            kernel_size=3,
+            stride=stride,
+            padding=0,
+            groups=hidden_dim,
+            bias=False,
+        )
+        self.bn = nn.BatchNorm2d(num_features=hidden_dim, eps=self.config.norm_eps, momentum=self.config.norm_momentum)
+        if self.use_se:
+            self.se = TrillssonSqueezeExcitation(config=self.config, in_channels=in_channels, out_channels=hidden_dim)
+            self.project_conv = Conv2DSamePadding(
+                in_channels=hidden_dim, out_channels=out_channels, kernel_size=1, stride=1, padding=0, bias=False
+            )
+            self.project_bn = nn.BatchNorm2d(
+                num_features=out_channels, eps=self.config.norm_eps, momentum=self.config.norm_momentum
+            )
+        self.gradient_checkpointing = False
+
+    def forward(self, input_features):
+        if self.expand_ratio != 1:
+            hidden_states = self.expand_conv(input_features)
+            hidden_states = self.expand_bn(hidden_states)
+            hidden_states = self.activation(hidden_states)
+        else:
+            hidden_states = input_features
+        hidden_states = self.dwconv2(hidden_states)
+        hidden_states = self.bn(hidden_states)
+        hidden_states = self.activation(hidden_states)
+        if self.use_se:
+            hidden_states = self.se(hidden_states)
+            hidden_states = self.project_conv(hidden_states)
+            hidden_states = self.project_bn(hidden_states)
+            if self.identity:
+                if self.survival_probability:
+                    hidden_states = self.dropout(hidden_states)
```

Comment:
Probably unecessary if the dropout probability is 0

Reply:
I don't understand what you mean but dropout probability is no 0, it's calculated by this equation `survival_probability = self.config.drop_connect_rate * current_num_blocks / total_blocks`

Reply:
Just a nit, meaning that it should be 0 by default rather than `None`, which could remove the`if` condition but it is really a small nit, ignore it